### PR TITLE
Update EXP_STOP to EXP_DELETE

### PR DESCRIPTION
**Pull request by kusumachalasani**
Signed-off-by: kusumachalasani <kchalasa@redhat.com>

This PR has changes related to:
- Modify EXP_STOP to EXP_DELETE and changes to the function names.
- Update viewExperiments URL to delete the experiment and the plots of that experiment for EXP_DELETE operation.

### CODE REVIEW

File: src/gRPC/hpo_pb2_grpc.py
Comment (chandrams) [inline src/gRPC/hpo_pb2_grpc.py:33]: @kusumachalasani - We shouldn't update these files by hand rather generate them

Comment (kusumachalasani) [reply src/gRPC/hpo_pb2_grpc.py:33]: oh..ok! didn't know. Will check that. Thanks!

Comment (chandrams) [reply src/gRPC/hpo_pb2_grpc.py:33]: You can use the below cmd, which can be found in the readme as well - https://github.com/kruize/hpo/tree/main/src#readme

```
python3 -m grpc_tools.protoc -I ./gRPC/protos --python_out=./gRPC --grpc_python_out=./gRPC gRPC/protos/*.proto
```

**Comment by kusumachalasani** [comment]
@johnaohara Please review the changes.

File: requirements.txt
Comment (chandrams) [inline requirements.txt:7]: @kusumachalasani - Do we need to specify the version here? Without explicitly specifying version, doesn't it install the latest protobuf ?

Comment (kusumachalasani) [reply requirements.txt:7]: Its not updating if there is a previous version of protobuf.

Comment (johnaohara) [reply requirements.txt:7]: This is a question about dependency versioning and repeatable builds.  by not versioning dependencies, builds are not repeatable at different points in time, as newer versions are automatically resolved. This can introduce issues due to incompatibilities. There is currently an inconsistency in the python requirements file, between explicit and latest versions of dependencies.  I would suggest picking a standard for the project and applying it consistently.  My inclination would be to explicitly state versions, and then use something like dependabot to explicitly manage dependency versions

File: src/gRPC/hpo_pb2.py
Comment (johnaohara) [inline src/gRPC/hpo_pb2.py:20]: Was this file automatically generated with `grpc_tools.protoc `? it appears to have removed a lot of the protobuf configuration

**Review by johnaohara — CHANGES_REQUESTED** [review]
@kusumachalasani I appologize for the time taken to review this PR.

The `./src/README.MD` should be updated with the new grpc client usage prompt (https://github.com/kusumachalasani/hpo/tree/exp_stop/src#grpc-client)

The command description for the `--recommended` command line option should also be updated to reflect the function of the `--recommended` option (https://github.com/kruize/hpo/blob/f2c3cd94c44f04e889d12987ed5b32954d6ae921/src/grpc_client.py#L152)

File: src/gRPC/hpo_pb2.py
Comment (kusumachalasani) [reply src/gRPC/hpo_pb2.py:20]: Yes. It was generated automatically. With the updates to protobuf version, these changes were seen.

File: requirements.txt
Comment (kusumachalasani) [reply requirements.txt:7]: @johnaohara Yes. we do accept that it should be consistent and explicitly state versions for all the dependencies. We shall take this as part of next PR.

**Review by johnaohara — APPROVED** [review]
LGTM

**Comment by kusumachalasani** [comment]
> @kusumachalasani I appologize for the time taken to review this PR.
> 
> The `./src/README.MD` should be updated with the new grpc client usage prompt (https://github.com/kusumachalasani/hpo/tree/exp_stop/src#grpc-client)
> 
> The command description for the `--recommended` command line option should also be updated to reflect the function of the `--recommended` option (
> 
> https://github.com/kruize/hpo/blob/f2c3cd94c44f04e889d12987ed5b32954d6ae921/src/grpc_client.py#L152
> 
> )

Updated the changes.

**Comment by johnaohara** [comment]
@kusumachalasani I have approved the changes, but an approval is required from someone with write access to the project

File: design/API.md
Comment (dinogun) [inline design/API.md:129]: Can you elaborate here, what happens to an ongoing experiment or only if the experiment has completed

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**
